docs/topics/plotting.rst

```diff
@@ -23,7 +23,7 @@ displaying multi-band images as RGB and labeling the axes with proper geo-refere
 
 The first argument to ``show`` represent the data source to be plotted. This can be one of
 
-   * A ``RasterReader`` source instance
+   * A ``DatasetReader`` source instance
```

Comment:
@j08lue I'd like to see the abstract term `dataset` used here instead of a class name. The reason why: I don't want to lead users towards calling the constructors of these classes directly. I want users to almost always use `rasterio.open()` as a factory for `DatasetReader` or `DatasetWriter` instances. I think that standardizing on the abstract term `dataset`, like Python does with `file`, will help.

Reply:
I like the idea of defining a `dataset` abstraction, following `file`.

So I guess this applies to all high-level mentions of `DatasetReader` - such in first-class function docstrings. I will see if there are any low-level cases where it makes sense to keep the class name, but those would be very few exceptions.